docs/download-upload-clone/high-level-design.md

```diff
@@ -0,0 +1,56 @@
+# High Level Design
+
+* exporting a project exports its underlying scenarios
+* as the client agreed, pre-defined resources do not need to be described 
+  with unique identifier shared across possible instances of MarxanCloud 
+  (Databases) - if any shared resource with imported ID is not available, 
+  import should be rejected
+* each of the resource above should consider if it is allowed to be used 
+  (license issues). In case it cannot be used during export, system should ..
+  take under consideration
+* existing data converted from shapefiles should be converted back to 
+  shapefile format (currently the platform doesn't keep the uploaded files, 
+  rather consumes them)
+* each export is a zip file
+* each export includes information about its current version
+  * in particular, it may also include what was exported, when, by whom etc...
+* components included in the exported files:
+  * Project:
+    * Project metadata (name, etc.)
+    * Project planning area (GADM), if provided
+    * Project planning area (from shapefile), if provided
+    * Project planning grid config (shape + unit area) and actual unit geometries, if generated
+    * Project planning grid (from shapefile), if provided
+  * Each scenario:
+    * Scenario metadata (name, etc.)
+    * IUCN categories for protected areas (note: if src and dest instances differ, and they use different WDPA versions, resulting protected area set may be different)
+    * Protected area geometries from shapefile
+    * Threshold for protected areas
+    * Features geometries for any user-supplied ones
+    * Feature specification (dependency on platform-wide features)
+    * Cost data per planning unit
+    * Lock status (default, unstated, locked-in, locked-out)
+    * Marxan settings (iterations, BLM, all advanced settings - input.dat in 
+      general)
+    * scenario run results (with option to not export or import them)
+* when declaring file to import, system should accept the options to:
+  * run scenarios with settings as per normal run once import is finished, 
+    as long as the solutions are not included in the imported file 
+* export operation is asynchronous, should block the usage of project/scenarios
```

Comment:
also, we shouldn't start an export operation if there are any running jobs for the project or any of its scenarios.

likely also if there are failed jobs (garbage in, garbage out - or rather in case of exports or clones, garbage out, garbage in)...

Reply:
@hotzevzl this may be tricky. If user already uploaded (for example) different cost surface (which failed), effectively previous one MAY still be valid.

As checking this may not be trivial... not sure if we should include such requirement? May it be worth discussing with client?

Reply:
blocking export added: https://github.com/Vizzuality/marxan-cloud/pull/551/commits/95ef5963ec984ade45f24305b88bb86ea0162049

Reply:
yep, that's correct, as you say - if users have solutions in the archive being imported and they want to preserve them, we need to make sure Marxan is not run after importing the archive